### PR TITLE
Selection tags mnrva 786

### PR DESCRIPTION
**Pull request by pras7430**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-786

 ## Description:
1. Tag dropdown added to left nav
2.  Test cases and changed

 ## Testing:

 ## Screenshots:
(if applicable)

### CODE REVIEW

File: src/app/_features/visualize/components/selectNav/metric-group/metric-dropdown.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MetricDropdownComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+  it('should add drown down create', () => {
```
Comment (FarajiA) [inline src/app/_features/visualize/components/selectNav/metric-group/metric-dropdown.component.spec.ts:25]: These 3 tests have all the same `it` statement but I can see they test different functionality.

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.spec.ts [hunk 2]
```diff
@@ -82,44 +83,54 @@ describe('VisualizePage', async() => {
...
-    component.dismissMetric(routes[1].queryParams.metric);
+    component.disMissedMetric(routes[1].queryParams.metric);
     expect(component.metricPillSet.size).toEqual(0);
+  }) 
+  it("should remove tags pills", () =>{
+    component.disMissedTag(routes[1].queryParams.tags);
+    expect(component.tagPillSet.size).toEqual(0);
+  })
+  
+  it("should change metrics", () =>{
```
Comment (FarajiA) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.spec.ts:108]: Same here. These `it` statements are identical

File: src/app/_mocks/metrics/metrics.json [hunk 3]
```diff
@@ -45,5 +94,21 @@
         "total_rw",
         "devices_ro",
         "devices_rw"
-    ]
+    ],
+    "tags":[{"metricGroup": "network",
+        "monitoring_system": "ZENOSS",
```
Comment (FarajiA) [inline src/app/_mocks/metrics/metrics.json:99]: For these let's not use real Acct numbers, device numbers or UUID numbers.

File: src/app/_services/metrics/metrics.service.ts [hunk 4]
```diff
@@ -52,7 +52,7 @@ export class MetricsService {
   */
   getMetricList(groupName:string) {
     httpOptions.headers['X-Tenant']=this.portalDataService.portalData.domainId;
-    return this.http.get<[string]>(`${this.metricsURL}/metadata/metricNames?group=${groupName}`,{ headers:httpOptions.headers,
+    return this.http.get<[string]>(`${this.metricsURL}/metadata/metricNames`,{ headers:httpOptions.headers,
```
Comment (FarajiA) [inline src/app/_services/metrics/metrics.service.ts:55]: This should be `string[]`?

File: src/app/_services/metrics/metrics.service.ts [hunk 5]
```diff
@@ -81,5 +81,20 @@ export class MetricsService {
       );
   }
 
+  getTagsList(data:any){
+    httpOptions.headers['X-Tenant']=this.portalDataService.portalData.domainId;
+    return this.http.get<[string]>(`${this.metricsURL}/metadata/tags`,{ headers:httpOptions.headers,
```
Comment (FarajiA) [inline src/app/_services/metrics/metrics.service.ts:86]: `string[]`?

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts [hunk 6]
```diff
@@ -39,97 +44,183 @@ export class VisualizePage {
...
+    this.privatemtrsrvc.getTagsList(para).subscribe((d) => {
+      d.forEach((it) => {
+        Object.entries(Object.assign({}, ...it)).forEach(a => {
+          this.tags = this.tags.add(a.join("="))
+        });
+      });
+    })
+  }
+  /*==============================================================================*/
+  /*===============================DropDown change event========================*/
```
Comment (FarajiA) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:175]: Nice. I like the /*==========*/ for code sections. I'll do the same

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts [hunk 7]
```diff
@@ -39,97 +44,183 @@ export class VisualizePage {
...
+
+  /**
+   * Metric Group Change
+   * @param changedGroup 
+   */
+  metricGroupChange(changedGroup) {
+    this.reset();
+    this.groupPillSet.add(changedGroup)
+    this.addGroupinQuery();
+    //this.getlistOfMetric(changedGroup);
```
Comment (FarajiA) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:185]: We can remove.

**Review by FarajiA — CHANGES_REQUESTED** [review]
Looks like we'll have to see why the GCP Build is failing.

**Review by FarajiA — APPROVED** [review]

File: package.json [hunk 8]
```diff
@@ -24,8 +24,9 @@
     "test:unit": "npm run mocksetup && ng test --code-coverage --project=minerva -c=mock --watch=false",
     "test:unit-ci": "ng test --project=minerva -c=mock --watch=false",
     "test:e2e": "npm run mocksetup && ng e2e --project=minerva -c=mock",
-    "test:e2e-ci": "webdriver-manager update --standalone false --gecko false --versions.chrome 87.0.4280.88 && ng e2e --project=minerva -c=mock",
+    "test:e2e-ci": "webdriver-manager update --standalone false --gecko false --versions.chrome 91.0.4472.77 && ng e2e --project=minerva -c=mock",
```
Comment (lovedeep92) [inline package.json:27]: package.json changes not need to be added to devel I think!

File: src/app/_features/visualize/components/selectNav/metric-group/metric-dropdown.component.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  @Output() dismissPill: EventEmitter<any> = new EventEmitter();
+
+  @Input() pills = new Set();
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  defaultgroup(item) {
+    if (!!this.defalutSelection && this.defalutSelection === item)
```
Comment (lovedeep92) [inline src/app/_features/visualize/components/selectNav/metric-group/metric-dropdown.component.ts:23]: it should be default, not defalut !!

**Pull request merged**